internal/webhook/v1alpha1/mariadb_webhook.go

```diff
@@ -44,10 +45,14 @@ func (d *MariaDBCustomDefaulter) Default(ctx context.Context, obj runtime.Object
 	}
 	mariadblog.V(1).Info("Defaulting for MariaDB", "name", mariadb.GetName())
 
-	if mariadb.Spec.Replication != nil && mariadb.Spec.Replication.Enabled {
+	env, err := environment.GetOperatorEnv(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting the environment: %v", err)
+	}
+
+	if mariadb.Replication().Enabled {
```

Comment:
We should align the implementation with Galera, and rely on `IsReplicationEnabled` for this cases.

Reply:
I think incrementally doing refactors is way better than making any assumptions. This is a big one, so it makes even more sense!